### ISSUE TITLE
get_components inconsistent return type

### ISSUE THREAD

**Issue by jd-lara** [NREL/PowerSystems.jl#219]
There is some inconsistent behavior in the `get_components` function for a concrete system. 

```Julia
include(joinpath(base_dir,"data/data_5bus_pu.jl"));
sys5 = PSY.System(nodes5, thermal_generators5, loads5, branches5, nothing,  100.0, forecasts5, nothing, nothing);

#Concrete System type
devices = PSY.get_components(PSY.ThermalDispatch, c_sys5);
typeof(devices)
> Base.Iterators.Take{Array{ThermalDispatch,1}} 

#Abstract Type 
devices = PSY.get_components(PSY.ThermalGen, c_sys5);
typeof(devices)
> Base.Iterators.Flatten{Base.Generator{Array{DataType,1},getfield(PowerSystems, Symbol("##84#87")){ConcreteSystem}}}
```

This is not consistent with the return types in `get_components`

**Comment by daniel-thom** [comment]
I'm confused...isn't this what you just fixed @jd-lara ?

**Comment by jd-lara** [comment]
The change didn't make the fix as planned. 

The current version of the code in `dev` still returns inconsistent iterators.

**Comment by jd-lara** [comment]
resolved in #206
